### ISSUE TITLE
Wrong HTTP Response codes in official docs

### ISSUE THREAD

**Issue by haasal** [tiangolo/fastapi#4975]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# I would have to copy the whole example from the doc which wouldn't help in any way so I will just outline the problematic parts:

# Should be 403 without the WWW-Authenticate Header

    for scope in security_scopes.scopes:
        if scope not in token_data.scopes:
            raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail="Not enough permissions",
                headers={"WWW-Authenticate": authenticate_value},
            )

# Should't this be a 401?

user = authenticate_user(fake_users_db, form_data.username, form_data.password)
    if not user:
        raise HTTPException(status_code=400, detail="Incorrect username or password")

# Basically the two above but in different places in the docs. See below
```

### Description

I was following the docs and noticed that many of the HTTP error response code where wrong (Or so do I think...).

## Locations

- [Here](https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/#global-view) and in all code "widgets" below I'm pretty sure the 401 should be a 403 like mentioned above. And I'm not sure but I believe the 400 should be a 401.
- 400 is probably fine for the inactive user.
- The errors are also inconsistent between doc pages as [here](https://fastapi.tiangolo.com/advanced/security/http-basic-auth/) 401 is used correctly.
- [Here](https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/) we are using 400 for unauthenticated users again.
- I didn't look over everything but these where the ones I saw.

### 401 -> 403

401 might be called Unauthorized but this is the problem here. It's used for invalid credentials or invalid authentication method where the server doesn't yet recognize the client.
But when we are checking for scope permission the server already knows who the client is and actively refuses to authorize it. Thus *403 FORBIDDEN*!
It doesn't make sense giving a *WWW-Authenticate* header if it changes nothing about the tokens permissions.

### 400 -> 401

I don't really understand why 400 was used in some places in the first place in case of failed authentication. If you use 400 
you have to give a detailed explanation about the error to make it clear how the client messed up. It would be more fitting to use 401 so the whole situation is easier to understand to the client.

## Links to sources

- [This](https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses) is the most comprehendable explanation.
- For convenience links to MDN docs: [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) and [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)

### Operating System

Linux

### Operating System Details

I use arch btw.

### FastAPI Version

xxx

### Python Version

xxx

### Additional Context

We definately need an issue form for doc improvements/bugs.

**Comment by JarroVGIT** [comment]
I tend to agree with you, worth updating the docs for. Might create a pull request if I find some time :)

**Comment by haasal** [comment]
I would really like to make the pull request if that's alright with you :). This would be the first time contributing to open source for me.

**Comment by JarroVGIT** [comment]
By all means!

On 6 Jun 2022, at 11:46, Alexander Haas ***@***.******@***.***>> wrote:



I would really like to make the pull request if that's alright with you :). This would be the first time contributing to open source for me.
